tests/cupy_tests/linalg_tests/test_product.py

```diff
@@ -377,6 +377,16 @@ def test_zerodim_kron(self, xp, dtype):
         b = testing.shaped_arange((4, 5), xp, dtype)
         return xp.kron(a, b)
 
+    @testing.numpy_cupy_allclose()
+    def test_kron_accepts_numbers_as_arguments(self, xp):
```

Comment:
We prefer using `@pytest.mark.parametrize` to `subTest` you suggested. Can I ask you to have it here? Thanks!

Reply:
I should be able to do this today. (Initially I tried an inner pytest decorator, but it had references to `xp.kron` within it, which failed.) With this refactoring, should I make `@pytest.mark.parametrize` the outer decorator? This seems consistent with other tests.

Reply:
Outer one is fine.

Reply:
I gave a try, and whether the pytest decorator is outer or inner, it doesn't seem to be passing its arguments to the test function. For example, as an outer decorator:
```python
    @pytest.mark.parametrize(
        "a, b", [
            (2, 3.0),
            (2, [[0, -1j / 2], [1j / 2, 0]]),
            ([[0, -1j / 2], [1j / 2, 0]], 2)
        ]
    )
    @testing.numpy_cupy_allclose()
    def test_kron_accepts_numbers_as_arguments(self, a, b, xp):
        args = [xp.array(arg) if type(arg) == list else arg for arg in [a, b]]
        print('converted args are ', args)   # print debug to be deleted before committing
        return xp.kron(*args)
```
results in the error
```
TypeError: TestProduct.test_kron_accepts_numbers_as_arguments() missing 2 required positional arguments: 'a' and 'b'
```

Full error below:
```
tests\cupy_tests\linalg_tests\test_product.py:379 (TestProduct.test_kron_accepts_numbers_as_arguments)
..\..\..\cupy\testing\_loops.py:273: in test_func
    _call_func_numpy_cupy(
..\..\..\cupy\testing\_loops.py:94: in _call_func_numpy_cupy
    cupy_result, cupy_error = _call_func_cupy(
..\..\..\cupy\testing\_loops.py:70: in _call_func_cupy
    result, error = _call_func(impl, args, kw)
..\..\..\cupy\testing\_loops.py:51: in _call_func
    raise e
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

impl = <function TestProduct.test_kron_accepts_numbers_as_arguments at 0x00000284BDFCD4E0>
args = (<cupy_tests.linalg_tests.test_product.TestProduct testMethod=test_kron_accepts_numbers_as_arguments>,)
kw = {'xp': <module 'cupy' from 'D:\\Documents\\GitHub\\cupy\\cupy\\__init__.py'>}

    def _call_func(impl, args, kw):
        # Note that `_pytest.outcomes.Skipped` is derived from BaseException.
        exceptions = Exception,
        if _is_pytest_available:
            exceptions += _pytest.outcomes.Skipped,
    
        try:
>           result = impl(*args, **kw)
E           TypeError: TestProduct.test_kron_accepts_numbers_as_arguments() missing 2 required positional arguments: 'a' and 'b'

..\..\..\cupy\testing\_loops.py:45: TypeError
```

Reply:
Ah, `@pytest.mark.parametrize` does not live in a `unittest.TestCase`'s subclass. Would you also make `TestProduct` class not deriving from `unittest.TestCase`? pytest needs no base testcase class.

Reply:
I moved the test into its own `TestKron` class, allowing `@pytest.mark_parametrize` to be used. Note the double commit because I almost forgot to remove my print debug statement. 🤦🏽‍♂️